src/kohlrahbi/conditions/__init__.py

```diff
@@ -2,36 +2,57 @@
 This module contains the function to write the collected conditions to a json file.
 """
 
-import json
 from pathlib import Path
 
+import docx  # type: ignore[import-untyped]
+from maus.edifact import EdifactFormat, EdifactFormatVersion, get_format_of_pruefidentifikator
+
+from kohlrahbi.ahb import get_pruefi_to_file_mapping
+from kohlrahbi.ahbtable.ahbcondtions import AhbConditions
+from kohlrahbi.ahbtable.ahbpackagetable import AhbPackageTable
 from kohlrahbi.logger import logger
+from kohlrahbi.read_functions import get_all_conditions_from_doc
+
 
+def find_all_files_from_all_pruefis(pruefi_to_file_mapping: dict[str, str]) -> dict[EdifactFormat, list[str]]:
+    """takes list of all pruefis with according files and returns a dict edifactformat-> list(filepaths)"""
+    format_to_files_mapping: dict[EdifactFormat, list[str]] = {}
+    for pruefi, filename in pruefi_to_file_mapping.items():
+        if not filename:
+            raise ValueError(f"No file provided for pruefi {pruefi}")
+        if format_to_files_mapping.get(get_format_of_pruefidentifikator(pruefi)) is None:
+            format_to_files_mapping[get_format_of_pruefidentifikator(pruefi)] = [filename]
+        elif filename not in format_to_files_mapping[get_format_of_pruefidentifikator(pruefi)]:
+            format_to_files_mapping[get_format_of_pruefidentifikator(pruefi)].append(filename)
```

Comment:
I guess we can introduce here a variable to avoid calling the `get_format_of_pruefidentifikator(pruefi)` function every time

Reply:
0e470fcbb9355457a17d7139c30efcfc1223de73